src/lib/scheduler/worker.cpp

```diff
@@ -44,12 +45,20 @@ void Worker::operator()() {
 
   _set_affinity();
 
+  std::mutex lock;
+  std::unique_lock<std::mutex> unique_lock(lock);
+  bool queues_are_empty;
+
   while (CurrentScheduler::get()->active()) {
-    _work();
+    queues_are_empty = _work();
+    if (queues_are_empty) {
+      // Wait for new task pushed to the current node's task queue or perform work stealing after 100 µs.
+      _queue->new_task.wait_for(unique_lock, std::chrono::microseconds(100));
```

Comment:
I haven't worked with condition variables for a while, but are you sure that you are using them correctly?

> **std::condition_variable::wait_for**
> [...]
> Calling this function if lock.mutex() is not the same mutex as the one used by all other threads that are currently waiting on the same condition variable is undefined behavior.

Reply:
You are right, I am using the wrong. I missed that note.

Reply:
Maybe also consider if you can reduce the time the lock is held.